### DISCUSSION TITLE
Corfu complete as type gets stuck

### DISCUSSION THREAD

**Discussion by eyJhb** [minad/corfu#529] [Q&A]
When I type and get auto completions it seems like it gets stuck while typing, e.g.

![image](https://github.com/user-attachments/assets/355a50ab-3efb-47f1-a47b-fcd19354ade6)

expected behavior is

![image](https://github.com/user-attachments/assets/c980ecd1-7dcd-43e6-8772-3d3a48b38809)

video showcasing it ![here](https://i.imgur.com/Ct6sXzk.mp4).

My configuration is as follows:

```el
(use-package corfu
  :init (global-corfu-mode 1)
  :config (progn
            (corfu-popupinfo-mode 1)
            (corfu-echo-mode 1)
            (setq 
             corfu-auto t
             corfu-auto-delay 0.1
             corfu-auto-prefix 1
             corfu-cycle t
             corfu-popupinfo-delay (cons 0.5 0.5))))

;; make corfu work in terminal as well
(use-package corfu-terminal
  :config (unless (display-graphic-p) (corfu-terminal-mode +1)))
```

Corfu - https://elpa.gnu.org/devel/corfu-1.5.0.20241105.213155.tar
Emacs - 30.0.92
Eglot - 1.17

**Comment by jdtsmith** [comment]
`corfu` is just a UI interface to the underlying completion system.  If you disable it and `M-Tab` at these same positions, are completions results shown?

**Reply by eyJhb** [reply]
I can get completion results with corfu as well, if I close the popup, and then start typing again. Which is why I'm not certain that it's a corfu or a eglot thing. 

Both screenshots are using corfu, on the same setup, etc. etc. the only difference is I clone the popup with C-g (forcefully, many times), and then started typing again.

**Reply by jdtsmith** [reply]
I'm not sure if that's answering the question: does the default completion UI give you the expected completion results at those times and locations?  If not, it's an eglot or lsp server issue, and you should follow along via `M-x eglot-events-buffer`.

**Reply by eyJhb** [reply]
So, I just now realize that the video I made was unavailable, which is kind of sad. I've made a new one here https://i.imgur.com/ViuNjRP.mp4

So the issue is as follows, if I start typing something, and I don't select the candidate and just continue typing, then it will not advance to a new prompt, with new candidates, as can be seen in the video.

If I however press enter, then continue to type, then the proper candidates will be shown.

Does this make sense? I have a _VERY_ hard time explaining it with the proper terminology.

**Reply by eyJhb** [reply]
Update. So I was running it on unstable Emacs, with the devel version specified above (in the initial comment), which didn't work. But running it with Emacs version 29.4 and stable corfu 1.5, it works as expected.

**Reply by eyJhb** [reply]
@jdtsmith @minad just a friendly ping, so that you're aware that it might be an issue on Emacs Unstable :) I'm unsure about how discussions work, RE notifications etc. 

(it was fixed by switching from unstable to stable)

**Reply by minad** [reply]
I use Emacs 30 without any issue. Which exact version do you have, Emacs git commit?

**Reply by jdtsmith** [reply]
This seems to me likely to be a problem with your completion backend.  I used to use `cape-capf-buster` to wrap eglot's completion-at-point function so that it would update its candidates on every change (company does this sort of thing by default).  Lately I haven't needed that, but perhaps that behavior has changed.  I'm on eglot 1.17.This seems to me likely to be a problem with your completion backend.  I used to use `cape-capf-buster` to wrap eglot's completion-at-point function so that it would update its candidates on every change (company does this sort of thing by default).  Lately I haven't needed that, but perhaps that behavior has changed.  I'm on eglot 1.17.

**Comment by eyJhb** [comment]
I'm now running Emacs 30.1, and I don't have any issues with corfu, so closing this now :)
